vnext/local-cli/src/config/dependencyConfig.ts

```diff
@@ -139,21 +139,20 @@ export function dependencyConfigWindows(
       sourceDir =
         'Error: Source dir is required if projects are specified, but it is null in react-native.config.';
     } else {
-      sourceDir = path.join(folder, userConfig.sourceDir);
+      sourceDir = path.join(folder, userConfig.sourceDir!);
     }
   } else if (!usingManualProjectsOverride) {
     // No manually provided projects, try to find sourceDir
     sourceDir = configUtils.findWindowsFolder(folder);
   }
 
   if (
```

Comment:
@jonthysell please let me know if this isn't the right fix. The previous logic leads to a compile error with this option beacause we seem to be able to let a null sourceDir past the conditionals, and could deref it soon later.